android/source/src/main/kotlin/com/gu/source/components/buttons/ButtonColours.kt

```diff
@@ -0,0 +1,183 @@
+package com.gu.source.components.buttons
+
+import com.gu.source.Source
+import com.gu.source.Source.Theme.Core
+import com.gu.source.Source.Theme.ReaderRevenue
+import com.gu.source.components.buttons.SourceButton.Priority.*
+import com.gu.source.daynight.AppColour
+import com.gu.source.presets.palette.*
+
+internal data class ButtonColours(
+    val border: AppColour,
+    val container: AppColour,
+    val content: AppColour,
+)
+
+@Suppress("CyclomaticComplexMethod", "LongMethod")
+/**
+ * Maps a [SourceButton.Priority] to a set of colours based on the theme.
+ */
+internal fun SourceButton.Priority.toColours(theme: Source.Theme) = when (theme) {
+    Core -> when (this) {
+        PrimaryOnWhite -> {
```

Comment:
Is this naming of PrimaryOnWhite what's used on the web? I'm wondering if PrimaryOnNeutral is a better name for apps because in dark mode this would be primary on black background. Consistency with web is probably going to be easier to maintain though, but it would be worth the design team thinking about because I imagine eventually they want dark mode support on web too? (At least for DCAR)

Reply:
I copied the naming that Simon had chosen in the news app using surface colour's visible name. The web uses different naming - `Brand` (OnBlue for us), `BrandAlt` (OnYellow, now removed), and `Default` (OnWhite for us).

I considered using the web naming but didn't go for it because it required knowledge of what brand and brand alt represented.

Both naming patterns - Brand or OnBlue - are also shaky because they only represent the light mode surface colour.